python/tk_multi_breakdown2/__init__.py

```diff
@@ -10,15 +10,49 @@
 
 from .api import BreakdownManager
 
+# For unit tests to access the app modules
+from .file_model import FileModel
+from .file_proxy_model import FileProxyModel
+from .file_history_model import FileHistoryModel
+
+try:
+    # Attempt to import the AppDialog
+    from .dialog import AppDialog
+except:
+    # Ignore import error for AppDialog so that the app works gracefully in batch modes
+    pass
+
 
 def show_dialog(app):
     """
     Show the main dialog ui
 
     :param app: The parent App
+    :type app: Application
+
+    :return: The dialog widget.
+    :rtype: AppDialog
     """
 
-    # defer imports so that the app works gracefully in batch modes
-    from .dialog import AppDialog
+    return app.engine.show_dialog("Scene Breakdown", app, AppDialog)
+
+
+def show_as_panel(app):
+    """
+    Return True if the App should be shown as a panel, instead of a dialog.
+
+    To be shown as a panel, the App should be set up to refresh on scene change events, such
+    as file open/new, reference files added/removed, etc. It does not make sense for the App
+    to be a dockable panel if it needs to be clsoed and re-opened to get updates.
```

Comment:
hey @staceyoue !
quick question here: maybe we'd like to be able to dock the app even if we don't have scene change events as you've also implemented the auto-refresh functionality?

Reply:
As per our discussion, docking the app will be allowed even if the scene change event callbacks have not been implemented since there is a manual refresh button now.